userspace/libsinsp/cri.hpp

```diff
@@ -808,6 +785,27 @@ inline bool cri_interface<api>::parse(const libsinsp::cgroup_limits::cgroup_limi
 					container.m_imagetag.c_str(), container.m_image.c_str(),
 					container.m_imagedigest.c_str());
 		}
+
+		/*
+		* The recent refactor makes full use of PodSandboxStatusResponse, removing the need to access pod sandbox containers
```

Comment:
@gnosek highlighting again that I am not introducing extra CRI API calls. They were already there before. 

In summary, the design is that for each container that is not a pod sandbox container you get a `ContainerStatsResponse` once and also a `PodSandboxStatusResponse` once and fully make use of the fact that you have the `PodSandboxStatusResponse`.

If we want to complicate things and cache `PodSandboxStatusResponse` info in pod sandbox containers so that the call is only done once for each pod is a different discussion. I would not recommend complicating things for now, but we could think about it in the future.